src/domain/UBGraphicsScene.cpp

```diff
@@ -138,6 +138,8 @@ qreal UBZLayerController::generateZLevel(QGraphicsItem *item)
         result =  generateZLevel(type);
     }
 
+    mAlreadyUsedZLevels.insert(result);
```

Comment:
Do we also have to delete a z level from the set when an item is deleted?

Reply:
Should we reassign everything every time a zValue is changed (e-g item with zLevel 99567 is removed, should item with zLevel 99566 chenge to zLevel 99567) ?

Or we don't do it, until item is grouped or user actively changes zLevel ?

Reply:
> Do we also have to delete a z level from the set when an item is deleted?

If I remove an object that was one of the first I put on the scene (so it's behind other objects), I don't want this new Item I'm adding to be put behind older objects. I we remove zLevel form the QSet, we should reasign other values, so there's no available "hole" in the set.